app/models/manageiq/providers/autosde/storage_manager/cloud_volume.rb

```diff
@@ -10,12 +10,24 @@ class ManageIQ::Providers::Autosde::StorageManager::CloudVolume < ::CloudVolume
 
   def self.raw_create_volume(ext_management_system, options = {})
     # @type [StorageService]
-    vol_to_create = ext_management_system.autosde_client.VolumeCreate(
-      :service => ext_management_system.storage_services.find(options["storage_service_id"]).ems_ref,
+    creation_hash = {
+      :service => "",
       :name    => options["name"],
       :size    => options["size"],
       :count   => options["count"]
-    )
+    }
+
+    if options['mode'] == 'Basic'
+      creation_hash[:service] = ext_management_system.storage_services.find(options["storage_service_id"]).ems_ref
```

Comment:
If `options["storage_service_id"]` is an array (which the lack of any `.to_a` down on [line 25](https://github.com/ManageIQ/manageiq-providers-autosde/pull/194/files#diff-54d13cf1217007ed4868559270a4b2861045c10b072508e2d5ae8d3e41f61f20R25) would indicate) then this `.find([id])` will return an array and `.ems_ref` will fail

Reply:
options["storage_service_id"] is a single uuid string value, passed from the non-multi-select field `storage_service_id`.  

options["storage_resource_id"] is an array of uuid strings.